src/Compilers/Core/Portable/Compilation/IStatement.cs

```diff
@@ -0,0 +1,463 @@
+        /// </summary>
+        CaseKind CaseKind { get; }
+    }
+
+    /// <summary>
+    /// Kinds of cases.
+    /// </summary>
+    public enum CaseKind
+    {
+        /// <summary>
+        /// Indicates case x in C# or Case x in VB.
+        /// </summary>
+        SingleValue,
+        /// <summary>
+        /// Indicates Case Is op x in VB.
+        /// </summary>
+        Relational,
+        /// <summary>
+        /// Indicates Case x To Y in VB.
+        /// </summary>
+        Range,
+        /// <summary>
+        /// Indicates default: in C3 or Case Else in VB.
+        /// </summary>
+        Default
+    }
+
+    /// <summary>
+    /// Represents case x in C# or Case x in VB.
+    /// </summary>
+    public interface ISingleValueCaseClause : ICaseClause
+    {
+        /// <summary>
+        /// Case value.
+        /// </summary>
+        IExpression Value { get; }
+        /// <summary>
+        /// Relational operator used to compare the switch value with the case value.
+        /// </summary>
+        RelationalOperationKind Equality { get; }
+    }
+
+    /// <summary>
+    /// Represents Case Is op x in VB.
+    /// </summary>
+    public interface IRelationalCaseClause : ICaseClause
+    {
+        /// <summary>
+        /// Case value.
+        /// </summary>
+        IExpression Value { get; }
+        /// <summary>
+        /// Relational operator used to compare the switch value with the case value. 
+        /// </summary>
+        RelationalOperationKind Relation { get; }
+    }
+
+    // Represents Case x To y in VB.
+    public interface IRangeCaseClause : ICaseClause
+    {
+        /// <summary>
+        /// Minimum value of the case range.
+        /// </summary>
+        IExpression MinimumValue { get; }
+        /// <summary>
+        /// Maximum value of the case range.
+        /// </summary>
+        IExpression MaximumValue { get; }
+    }
+
+    /// <summary>
+    /// Represents an if statement in C# or an If statement in VB.
+    /// </summary>
+    public interface IIf : IStatement
+    {
+        /// <summary>
+        /// Clauses of the if. For C# there is one clause per if, but for VB there can be multiple.
+        /// </summary>
+        ImmutableArray<IIfClause> IfClauses { get; }
+        /// <summary>
+        /// Else of the if statement.
+        /// </summary>
+        IStatement Else { get; }
+    }
+
+    /// <summary>
+    /// Represents a conditional clause of an if statement.
+    /// </summary>
+    public interface IIfClause
+    {
+        /// <summary>
+        /// Condition of the clause.
+        /// </summary>
+        IExpression Condition { get; }
+        /// <summary>
+        /// Body of the clause.
+        /// </summary>
+        IStatement Body { get; }
+    }
+
+    /// <summary>
+    /// Represents a C# while, for, foreach, or do statement, or a VB While, For, For Each, or Do statement.
+    /// </summary>
+    public interface ILoop : IStatement
```

Comment:
Shouldn't all implementations of IStatement have a suffix -Statement? ILoopStatement, IForStatement, etc. It would be more consistent with syntax APIs.